File: Tvornice, uređivač teksta/tiger.c
#include "myfactory.h"
#include <stdio.h>
#include <stdlib.h>

typedef char const* (*PTRFUN)();

typedef struct Tiger {
	PTRFUN* table;
	const char* name;
} Tiger;

char const* name(void *this) {
	return ((Tiger*)this)->name;
}

char const* greet() {
	return "rawr";
}

char const* menu() {
	return "sve oko sebe";
}

int size_of() {
	return sizeof(Tiger);
}

PTRFUN pfun[] = { name, greet, menu };

void construct(Tiger* memory, char const* name) {
	memory->name = name;
	memory->table = pfun;
}

void* create(char const* name) {
	struct Tiger* mal = malloc(size_of());
	construct(mal, name);
	return mal;
}

File: Tvornice, uređivač teksta/OOUP3/src/main/java/hr/fer/zemris/ooup/lab3/model/plugins/Tiger.java
package hr.fer.zemris.ooup.lab3.model.plugins;

import hr.fer.zemris.ooup.lab3.model.Animal.Animal;

public class Tiger extends Animal {

    private String animalName;

    public Tiger(String name) {
        this.animalName = name;
    }
    @Override
    public String name() {
        return this.animalName;
    }

    @Override
    public String greet() {
        return "raaawr";
    }

    @Override
    public String menu() {
        return "sve oko sebe";
    }
}
